apisix/plugins/log-rotate.lua

```diff
@@ -49,6 +50,7 @@ local default_logs
 local enable_compression = false
 local DEFAULT_ACCESS_LOG_FILENAME = "access.log"
 local DEFAULT_ERROR_LOG_FILENAME = "error.log"
+local SIGUSR1 = 10
```

Comment:
Because signal.signum may return nil, so we give it a default value when it's null, this function's doc says:
`
Maps the signal name specified to the system-specific signal number. Returns nil if the signal name is not known
`

Reply:
Another choice is avoiding to send kill signal when signal.signum returns nil.

Reply:
@qihaiyan 
Could you tell me which platform we need to support but doesn't have SIGUSR1?

Reply:
> @qihaiyan Could you tell me which platform we need to support but doesn't have SIGUSR1?

I don't know, but the signal.signum function's doc says `Returns nil if the signal name is not known`. 
If we use signal.signum's return value directly, a warn is reported with `Lua` plugin in vscode.

Reply:
The nil check has been removed.